lldb/source/Target/Process.cpp

```diff
@@ -6686,9 +6710,12 @@ Status Process::CalculateCoreFileSaveRanges(const SaveCoreOptions &options,
   if (err.Fail())
     return err;
 
-  if (ranges.empty())
+  if (ranges.IsEmpty())
     return Status("no valid address ranges found for core style");
 
+  // Sort the range data vector to dedupe ranges before returning.
+  ranges.Sort();
```

Comment:
This is where we should go over the ranges and break them up by permissions. Also if we wanted to support the user specifying large ranges that contain many regions, then we would break them up here as well.